src/test/rules/sortImportsRuleTests.ts

```diff
@@ -0,0 +1,221 @@
+import { Failure, Position, RuleTester, dedent } from './ruleTester';
+
+const ALPHA_ORDER_ERROR = (x, y) => `All imports of the same type must be sorted alphabetically. "${x}" must come before "${y}"`;
+const TYPE_ORDER_ERROR = (x, y) => `All imports of type "${x}" must occur before all imports of type "${y}"`;
+const MEMBER_SORT_ERROR = 'Member imports must be sorted alphabetically.';
+
+function expecting(errors: [string, number, number, number][]): Failure[] {
+  // message, line, startColumn, endColumn
+  return errors.map((err) => {
+    return {
+      failure: err[0],
+      startPosition: new Position(err[1], err[2]),
+      endPosition: new Position(err[1], err[3])
+    };
+  });
+}
+
+const ruleTester = new RuleTester('sort-imports');
+
+ruleTester.addTestGroup('valid', 'should pass ESLint valid tests', [
+  dedent`
+    import a from 'foo';
+    import b from 'bar';
+    import c from 'baz';`,
+  dedent`
+    import * as B from 'foo';
+    import A from 'bar';`,
+  dedent`
+    import * as B from 'foo';
+    import {a, b} from 'bar';`,
+  {
+    code:
+    dedent`
+      import A from 'bar';
+      import {b, c} from 'foo'`,
+    options: [{ 'member-syntax-sort-order': ['single', 'multiple', 'none', 'all', 'alias'] }]
+  },
+  dedent`
+    import {a, b} from 'bar';
+    import {c, d} from 'foo';`,
+  dedent`
+    import A from 'foo';
+    import B from 'bar';`,
+  dedent`
+    import A from 'foo';
+    import a from 'bar';`,
+  dedent`
+    import a, * as b from 'foo';
+    import c from 'bar';`,
+  dedent`
+    import 'foo';
+    import a from 'bar';`,
+  dedent`
+    import B from 'foo.js';
+    import a from 'bar';`,
+  {
+    code: dedent`
+      import a from 'foo';
+      import B from 'bar';`,
+    options: ['ignore-case']
+  },
+  {
+    code:
+    `import {a, b, c, d} from 'foo';`
```

Comment:
this line can be placed right next to code or simply remove the object and just place the string.